resource-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/resource/kafka/KafkaInitalizer.java

```diff
@@ -0,0 +1,156 @@
+
+        @Bean
+        @ConditionalOnMissingBean
+        static PropertiesConfigurer propertiesConfigurer(final ResourceLoader resourceLoader, final Environment environment) {
+            return new PropertiesConfigurer(resourceLoader, environment);
+        }
+
+        @Bean
+        KafkaContainer kafkaContainer() throws ExecutionException, InterruptedException {
+            createTopics(this.topics, kafka);
+            return kafka;
+        }
+
+        @Bean
+        @Primary
+        ActorSystem actorSystem(final PropertiesConfigurer props, final KafkaContainer kafka, final ConfigurableApplicationContext context) {
+            LOGGER.info("Starting Kafka with port {}", kafka.getFirstMappedPort());
+            return ActorSystem.create("actor-with-overrides", props.toHoconConfig(Stream.concat(
+                    props.getAllActiveProperties().entrySet().stream()
+                            .filter(kafkaPort -> !kafkaPort.getKey().equals("akka.discovery.config.services.kafka.endpoints[0].port")),
+                    Stream.of(new AbstractMap.SimpleEntry<>("akka.discovery.config.services.kafka.endpoints[0].port", Integer.toString(kafka.getFirstMappedPort()))))
+                    .collect(toMap(Map.Entry::getKey, Map.Entry::getValue))));
+        }
+
+        @Bean
+        @Primary
+        Materializer materializer(final ActorSystem system) {
+            return Materializer.createMaterializer(system);
+        }
+    }
+
+    // Serialiser for upstream test input
+    static class RequestSerializer implements Serializer<JsonNode> {
+        @Override
+        public byte[] serialize(final String s, final JsonNode userRequest) {
+            try {
+                return MAPPER.writeValueAsBytes(userRequest);
+            } catch (JsonProcessingException e) {
+                throw new SerializationFailedException("Failed to serialize " + userRequest.toString(), e);
+            }
+        }
+    }
+
+    // Deserialiser for downstream test output
+    static class ResponseDeserializer implements Deserializer<JsonNode> {
+        @Override
+        public JsonNode deserialize(final String s, final byte[] userResponse) {
+            try {
+                return MAPPER.readTree(userResponse);
+            } catch (IOException e) {
+                throw new SerializationFailedException("Failed to deserialize " + new String(userResponse), e);
+            }
+        }
+    }
+
+    // Deserialiser for downstream test error output
+    static class ErrorDeserializer implements Deserializer<AuditErrorMessage> {
+        @Override
+        public AuditErrorMessage deserialize(final String s, final byte[] auditErrorMessage) {
+            try {
+                return MAPPER.readValue(auditErrorMessage, AuditErrorMessage.class);
+            } catch (IOException e) {
+                throw new SerializationFailedException("Failed to deserialize " + new String(auditErrorMessage), e);
+            }
+        }
+    }
```

Comment:
This is unused as in the contract test you deseralise the error topic to string, not AuditErrorMessage. 
Either refactor the tests to deseralise into the object, or remove this deseraliser

Reply:
Removed this unused deserialiser.